### ISSUE TITLE
[Blacklist] Unhandled logs

### ISSUE THREAD

**Issue by keith** [supermarin/xcpretty#263]
On the [blacklist branch](https://github.com/supermarin/xcpretty/tree/blacklist) I've noticed some unhandled logs in our builds. I thought it would make sense to have an issue to track these.

**Comment by keith** [comment]
`ProcessProductPackaging`:

```
ProcessProductPackaging "" build/Build/Intermediates/Lyft.build/QA-iphonesimulator/Lyft.build/Lyft.app.xcent
    cd /Users/distiller/Lyft-iOS
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    

Entitlements:

{
    "application-identifier" = "<REDACTED>";
    "com.apple.developer.associated-domains" =     (
        "<REDACTED>",
        "<REDACTED>"
    );
    "com.apple.developer.siri" = 1;
    "com.apple.developer.ubiquity-kvstore-identifier" = "<REDACTED>";
    "com.apple.security.application-groups" =     (
        "<REDACTED>"
    );
}


    builtin-productPackagingUtility -entitlements -format xml -o /Users/distiller/Lyft-iOS/build/Build/Intermediates/Lyft.build/QA-iphonesimulator/Lyft.build/Lyft.app.xcent
```

**Comment by keith** [comment]
`CopyStringsFile`:

```
CopyStringsFile build/Build/Products/QA-iphonesimulator/Lyft.app/Base.lproj/InfoPlist.strings Lyft/Supporting\ Files/Base.lproj/InfoPlist.strings
    cd /Users/distiller/Lyft-iOS
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    builtin-copyStrings --validate --inputencoding utf-8 --outputencoding binary --outdir /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/Lyft.app/Base.lproj -- Lyft/Supporting\ Files/Base.lproj/InfoPlist.strings
```

**Comment by keith** [comment]
`CreateUniversalBinary`:

```
CreateUniversalBinary /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework/FBSDKLoginKit normal i386\ x86_64
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode-8.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo -create /Users/distiller/Lyft-iOS/build/Build/Intermediates/Pods.build/QA-iphonesimulator/FBSDKLoginKit.build/Objects-normal/i386/FBSDKLoginKit /Users/distiller/Lyft-iOS/build/Build/Intermediates/Pods.build/QA-iphonesimulator/FBSDKLoginKit.build/Objects-normal/x86_64/FBSDKLoginKit -output /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework/FBSDKLoginKit
```

**Comment by keith** [comment]
`GenerateDSYMFile`:

```
GenerateDSYMFile /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework.dSYM /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework/FBSDKLoginKit
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode-8.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework/FBSDKLoginKit -o /Users/distiller/Lyft-iOS/build/Build/Products/QA-iphonesimulator/FBSDKLoginKit/FBSDKLoginKit.framework.dSYM
```

**Comment by keith** [comment]
`CompileStoryboard`

```
CompileStoryboard Lyft/Resources/Storyboards\ &\ XIBs/Passenger/Map.storyboard
    cd /Users/distiller/Lyft-iOS
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    export XCODE_DEVELOPER_USR_PATH=/Applications/Xcode-8.1.app/Contents/Developer/usr/bin/..
    /Applications/Xcode-8.1.app/Contents/Developer/usr/bin/ibtool --errors --warnings --notices --module Lyft --output-partial-info-plist /Users/distiller/Lyft-iOS/build/Build/Intermediates/Lyft.build/QA-iphonesimulator/Lyft.build/Map-SBPartialInfo.plist --auto-activate-custom-fonts --target-device iphone --minimum-deployment-target 8.0 --output-format human-readable-text --compilation-directory /Users/distiller/Lyft-iOS/build/Build/Intermediates/Lyft.build/QA-iphonesimulator/Lyft.build /Users/distiller/Lyft-iOS/Lyft/Resources/Storyboards\ &\ XIBs/Passenger/Map.storyboard
```

**Comment by keith** [comment]
`CompileSwift`:

```
CompileSwift normal x86_64
    cd /Users/distiller/Lyft-iOS
    /Applications/Xcode-8.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c 
```

**Comment by supermarin** [comment]
Thanks for pasting examples :)
I've already implemented some of them on ✈️ , just pushed in branch. Will bump the lockfile when I address all of these

**Comment by joshlytle** [comment]
xcpretty is crashing on the following two lines consistently. I don't know why they were being generated by my build, and they were fixed by a computer restart, but it'd be good to correct the underlying issue:

2017-01-20 13:04:18.471 appledoc[43157:8803325] kCFURLVolumeIsAutomountedKey missing for file:///Volumes/HQSAN01/: Error Domain=NSCocoaErrorDomain Code=257 "The file “HQSAN01” couldn’t be opened because you don’t have permission to view it." UserInfo={NSURL=file:///Volumes/HQSAN01/, NSFilePath=/Volumes/HQSAN01, NSUnderlyingError=0x7f82f27b69e0 {Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied"}}
2017-01-20 13:04:19.308 appledoc[43157:8803178] ### generator: <GBDocSetInstallGenerator: 0x7f82f2aead70>, store: <GBStore: 0x7f82e3e07c50>, error: Documentation set was installed, but couldn't reload documentation within Xcode.

**Comment by supermarin** [comment]
@joshlytle this is not xcpretty crashing, this is xcodebuild stderr output.
If you're not getting any output after that, you either:
- don't have a working xcpretty in your PATH
- have xcodebuild stuck right on beginning

I suggest running xcodebuild command without xcpretty and figuring out what the raw output looks like, then adding back xcpretty when it's all working.

**Comment by joshlytle** [comment]
I didn't post the error log; I posted the last line that was fed through xcpretty which caused a crash in the Ruby xcpretty is running. It occasionally crashes on many such lines which can be generated when a build contains script steps, which could be doing anything and outputting anything.

Here's the output when xcpretty encounters the above log:

/Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/lib/xcpretty/parser.rb:422:in `===': invalid byte sequence in US-ASCII (ArgumentError)
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/lib/xcpretty/parser.rb:422:in `update_test_state'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/lib/xcpretty/parser.rb:299:in `parse'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/lib/xcpretty/formatters/formatter.rb:87:in `pretty_format'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/lib/xcpretty/printer.rb:19:in `pretty_print'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/bin/xcpretty:84:in `block in <top (required)>'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/bin/xcpretty:83:in `each_line'
	from /Library/Ruby/Gems/2.0.0/gems/xcpretty-0.2.4/bin/xcpretty:83:in `<top (required)>'
	from /usr/local/bin/xcpretty:23:in `load'
	from /usr/local/bin/xcpretty:23:in `<main>'

**Comment by supermarin** [comment]
@joshlytle please search the issues, this is a known problem. Let's keep this Issue ontopic.

tl;dr; you need to set locale to utf-8.
Please follow up in a separate issue

**Comment by joshlytle** [comment]
I apologize. I figured an issue entitled "unhandled logs" would be a good place to post unhandled logs. I will do additional digging. Thank you.

**Comment by sberrevoets** [comment]
Adding a few more:

```
SwiftCodeGeneration normal arm64 /Users/distiller/Driver-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/Pods.build/Beta-iphoneos/SnapKit.build/Objects-normal/arm64/ConstraintMakerPriortizable.bc
    cd /Users/distiller/Driver-iOS/Pods
    /Applications/Xcode-8.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c -primary-file /Users/distiller/Driver-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/Pods.build/Beta-iphoneos/SnapKit.build/Objects-normal/arm64/ConstraintMakerPriortizable.bc -target arm64-apple-ios8.0 -Xllvm -aarch64-use-tbi -O -module-name SnapKit -o /Users/distiller/Driver-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/Pods.build/Beta-iphoneos/SnapKit.build/Objects-normal/arm64/ConstraintMakerPriortizable.o -embed-bitcode -disable-llvm-optzns```

**Comment by sberrevoets** [comment]
```
SetMode u+w,go-w,a+rX /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FBSDKCoreKit.framework
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/Users/distiller/Library/Python/2.7/bin:/Users/distiller/Library/Python/2.7/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /bin/chmod -RH u+w,go-w,a+rX /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FBSDKCoreKit.framework
```

**Comment by sberrevoets** [comment]
```
SymLink /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/BuildProductsPath/Beta-iphoneos/libPhoneNumber-iOS/libPhoneNumber_iOS.framework /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libPhoneNumber_iOS.framework
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/Users/distiller/Library/Python/2.7/bin:/Users/distiller/Library/Python/2.7/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /bin/ln -sfh /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libPhoneNumber_iOS.framework /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/BuildProductsPath/Beta-iphoneos/libPhoneNumber-iOS/libPhoneNumber_iOS.framework
```

**Comment by sberrevoets** [comment]
```
Strip /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libPhoneNumber_iOS.framework/libPhoneNumber_iOS
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/Users/distiller/Library/Python/2.7/bin:/Users/distiller/Library/Python/2.7/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode-8.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -S /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/libPhoneNumber_iOS.framework/libPhoneNumber_iOS
```

**Comment by supermarin** [comment]
Thanks Scott

**Comment by supermarin** [comment]
Adding `SetOwnerAndGroup` support for as well

```
SetOwnerAndGroup distiller:staff /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/watchos/LambdaKit.framework
    cd /Users/distiller/Lyft-iOS/Pods
    export PATH="/Applications/Xcode-8.1.app/Contents/Developer/Platforms/WatchOS.platform/Developer/usr/bin:/Applications/Xcode-8.1.app/Contents/Developer/usr/bin:/Users/distiller/Library/Python/2.7/bin:/Users/distiller/Library/Python/2.7/bin:/usr/local/bin:/usr/local/lib/ruby/gems/2.3.0/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /usr/sbin/chown -RH distiller:staff /Users/distiller/Lyft-iOS/build/Build/Intermediates/ArchiveIntermediates/Lyft/IntermediateBuildFilesPath/UninstalledProducts/watchos/LambdaKit.framework
```

**Comment by keith** [comment]
Looks like beside `ProcessProductPackaging` this issue might be done?

**Comment by keith** [comment]
Last one here: https://github.com/supermarin/xcpretty/pull/324
